homeassistant/components/linky/sensor.py

```diff
@@ -29,29 +50,64 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     username = config[CONF_USERNAME]
     password = config[CONF_PASSWORD]
     timeout = config[CONF_TIMEOUT]
+    monitored_conditions = config[CONF_MONITORED_CONDITIONS]
+
+    LinkyAccount(hass, add_entities, username, password, timeout,
+                 monitored_conditions)
 
-    from pylinky.client import LinkyClient, PyLinkyError
-    client = LinkyClient(username, password, None, timeout)
-    try:
-        client.login()
-        client.fetch_data()
-    except PyLinkyError as exp:
-        _LOGGER.error(exp)
-        client.close_session()
-        return
 
-    devices = [LinkySensor('Linky', client)]
-    add_entities(devices, True)
+class LinkyAccount:
+    """Device manager for demo geolocation events."""
+
+    def __init__(self, hass, add_entities, username, password, timeout,
+                 monitored_conditions):
+        """Initialise the demo geolocation event manager."""
+        self.hass = hass
+        self._client = LinkyClient(username, password, None, timeout)
+
+        try:
+            self._client.login()
+            self._client.fetch_data()
+        except PyLinkyError as exp:
+            _LOGGER.error(exp)
+            self._client.close_session()
+            return
+
+        self.__devices = []
+        for condition in monitored_conditions:
+            self.__devices.append(
+                LinkySensor(SENSORS[condition][SENSORS_INDEX_LABEL],
+                            self._client,
+                            SENSORS[condition][SENSORS_INDEX_SCALE],
+                            SENSORS[condition][SENSORS_INDEX_WHEN]))
+
+        add_entities(self.__devices, True)
+        track_time_interval(hass, self.update_linky_data, SCAN_INTERVAL)
+
+    def update_linky_data(self, event_time):
+        """Fetch new state data for the sensor."""
+        try:
+            self._client.fetch_data()
+        except PyLinkyError as exp:
+            _LOGGER.error(exp)
+            self._client.close_session()
+            self._client.login()
+            return
+
+        _LOGGER.debug(json.dumps(self._client.get_data(), indent=2))
 
 
 class LinkySensor(Entity):
     """Representation of a sensor entity for Linky."""
 
-    def __init__(self, name, client):
+    def __init__(self, name, client, scale, when):
         """Initialize the sensor."""
         self._name = name
         self._client = client
+        self._scale = scale
+        self._when = when
         self._state = None
+        self._last_updated = None
```

Comment:
Remove this. The [state object](https://www.home-assistant.io/docs/configuration/state_object/) also has this information.

Reply:
Can we add this info to the device_state_attributes property ?